### ISSUE TITLE
[BUG] HuggingFaceEmbeddings: ValidationError: 1 validation error for HuggingFaceHubEmbeddings encode_kwargs

### ISSUE THREAD

**Issue by tlevyaleia** [chatchat-space/Langchain-Chatchat#3868]
**Problem Description**
I am using a remote embedding model started with [text-embeddings-inference](https://huggingface.co/docs/text-embeddings-inference/index).
I want to use this embedding in langchain and set the {'batch_size': 16} as encode_kwargs.
I get the following error when initializing the HuggingFaceHubEmbeddings object:
```
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In[17], line 5
      3 #encode_kwargs = {'normalize_embeddings': False}
      4 encode_kwargs = {'batch_size': 16}
----> 5 embedding = HuggingFaceHubEmbeddings(model="my_url", huggingfacehub_api_token="xxx", encode_kwargs=encode_kwargs)

File /aleia/jplab/.venv/lib/python3.9/site-packages/pydantic/v1/main.py:341, in BaseModel.__init__(__pydantic_self__, **data)
    339 values, fields_set, validation_error = validate_model(__pydantic_self__.__class__, data)
    340 if validation_error:
--> 341     raise validation_error
    342 try:
    343     object_setattr(__pydantic_self__, '__dict__', values)

ValidationError: 1 validation error for HuggingFaceHubEmbeddings
encode_kwargs
  extra fields not permitted (type=value_error.extra)
 
```

**复现问题的步骤 / Steps to Reproduce**
Install sentence-transformers (2.2.2) and langchain (0.1.16) packages
Running the following code:
```
from langchain_community.embeddings import HuggingFaceHubEmbeddings

#encode_kwargs = {'normalize_embeddings': False}
encode_kwargs = {'batch_size': 16}
embedding = HuggingFaceHubEmbeddings(model="my_url", huggingfacehub_api_token="xxx", encode_kwargs=encode_kwargs)

```

**预期的结果 / Expected Result**
I expect the embedding object to be initialized without raising an error

**实际结果 / Actual Result**
ValidationError raised.

**环境信息 / Environment Information**
-langchain: 0.1.16
-No docker deployment
-Python version: Python 3.9.19
-pip list:
```
Package                       Version
----------------------------- -----------
a2wsgi                        1.4.0
aiobotocore                   1.4.2
aiohttp                       3.8.6
aioitertools                  0.11.0
aiosignal                     1.3.1
alabaster                     0.7.13
aleialib                      0.1.91
aleiamlops                    0.1.1
aleiamodel                    0.1.14
alembic                       1.12.0
annotated-types               0.6.0
anyio                         4.0.0
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
arrow                         1.3.0
asgiref                       3.5.0
asttokens                     2.4.0
async-generator               1.10
async-lru                     2.0.4
async-timeout                 4.0.3
attrs                         23.1.0
Babel                         2.13.0
backcall                      0.2.0
beautifulsoup4                4.12.2
bleach                        6.1.0
boto3                         1.17.106
botocore                      1.20.106
cachetools                    5.3.1
certifi                       2023.5.7
certipy                       0.1.3
cffi                          1.16.0
cfgv                          3.4.0
chardet                       3.0.4
charset-normalizer            3.3.0
click                         8.0.4
cloudpickle                   2.2.1
comm                          0.1.4
coverage                      7.3.2
cryptography                  3.4.8
cycler                        0.12.1
dataclasses-json              0.6.4
debugpy                       1.8.0
decorator                     5.1.1
defusedxml                    0.7.1
dill                          0.3.6
distlib                       0.3.7
dnspython                     1.16.0
docutils                      0.20.1
editables                     0.5
et-xmlfile                    1.1.0
exceptiongroup                1.1.3
executing                     2.0.0
expects                       0.9.0
faiss-cpu                     1.8.0
fastjsonschema                2.18.1
filelock                      3.12.4
flaky                         3.7.0
Flask                         1.1.2
Flask-Cors                    3.0.10
Flask-OpenTracing             1.1.0
flatbuffers                   1.12
fqdn                          1.5.1
frozenlist                    1.4.0
fsspec                        2024.3.1
gcsfs                         2021.11.0
google-api-core               2.12.0
google-auth                   2.23.3
google-auth-oauthlib          1.1.0
google-cloud-core             2.3.3
google-cloud-storage          2.12.0
google-crc32c                 1.5.0
google-resumable-media        2.6.0
googleapis-common-protos      1.61.0
greenlet                      1.1.2
grpcio                        1.62.1
grpcio-opentracing            1.1.4
grpcio-reflection             1.34.1
gunicorn                      20.1.0
h11                           0.13.0
h5py                          3.9.0
hatch                         1.7.0
hatchling                     1.18.0
hdf5plugin                    4.2.0
httpcore                      0.18.0
httpx                         0.25.0
huggingface-hub               0.22.2
hybridcontents                0.4.0
hyperlink                     21.0.0
identify                      2.5.30
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.8.0
importlib-resources           6.1.0
iniconfig                     2.0.0
ipykernel                     6.25.2
ipython                       8.12.3
ipython-genutils              0.2.0
isodate                       0.6.1
isoduration                   20.11.0
itsdangerous                  1.1.0
jaeger-client                 4.4.0
jaraco.classes                3.3.0
jedi                          0.19.1
jeepney                       0.8.0
Jinja2                        2.11.3
jmespath                      0.10.0
joblib                        1.1.0
json5                         0.9.14
jsonpatch                     1.33
jsonpointer                   2.4
jsonschema                    3.2.0
jsonschema-spec               0.2.4
jsonschema-specifications     2023.7.1
jupyter_client                8.4.0
jupyter_core                  5.4.0
jupyter-events                0.6.3
jupyter-lsp                   2.2.0
jupyter_server                2.8.0
jupyter_server_terminals      0.4.4
jupyter-telemetry             0.1.0
jupyterhub                    4.0.2
jupyterlab                    4.0.7
jupyterlab-pygments           0.2.2
jupyterlab_server             2.25.0
keyring                       24.2.0
kiwisolver                    1.4.5
langchain                     0.1.16
langchain-community           0.0.34
langchain-core                0.1.45
langchain-text-splitters      0.0.1
langsmith                     0.1.50
lazy-object-proxy             1.9.0
Mako                          1.2.4
markdown-it-py                3.0.0
MarkupSafe                    1.1.1
marshmallow                   3.21.1
matplotlib                    3.4.3
matplotlib-inline             0.1.6
mdurl                         0.1.2
mistune                       3.0.2
more-itertools                10.1.0
mpmath                        1.3.0
multidict                     6.0.4
mypy-extensions               1.0.0
nbclient                      0.8.0
nbconvert                     7.9.2
nbformat                      5.9.2
nbval                         0.10.0
nest-asyncio                  1.5.8
networkx                      3.2.1
nltk                          3.8.1
nodeenv                       1.8.0
notebook                      7.0.6
notebook_shim                 0.2.3
numpy                         1.24.4
nvidia-cublas-cu12            12.1.3.1
nvidia-cuda-cupti-cu12        12.1.105
nvidia-cuda-nvrtc-cu12        12.1.105
nvidia-cuda-runtime-cu12      12.1.105
nvidia-cudnn-cu12             8.9.2.26
nvidia-cufft-cu12             11.0.2.54
nvidia-curand-cu12            10.3.2.106
nvidia-cusolver-cu12          11.4.5.107
nvidia-cusparse-cu12          12.1.0.106
nvidia-nccl-cu12              2.19.3
nvidia-nvjitlink-cu12         12.4.127
nvidia-nvtx-cu12              12.1.105
oauthlib                      3.2.2
odfpy                         1.4.1
openapi-core                  0.18.1
openapi-schema-validator      0.6.2
openapi-spec-validator        0.6.0
openpyxl                      3.1.2
opentracing                   2.4.0
orjson                        3.10.1
overrides                     7.4.0
packaging                     23.2
pamela                        1.1.0
pandas                        1.3.5
pandocfilters                 1.5.0
parse                         1.19.1
parso                         0.8.3
pathable                      0.4.3
pathspec                      0.11.2
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        10.1.0
pip                           24.0
platformdirs                  3.11.0
pluggy                        1.3.0
pre-commit                    3.5.0
prometheus-client             0.8.0
prompt-toolkit                3.0.39
protobuf                      3.20.3
psutil                        5.9.6
psycopg2-binary               2.9.9
ptyprocess                    0.7.0
pure-eval                     0.2.2
pyarrow                       12.0.1
pyasn1                        0.5.0
pyasn1-modules                0.3.0
pycparser                     2.21
pycryptodome                  3.10.1
pydantic                      2.7.1
pydantic_core                 2.18.2
pyenchant                     3.2.2
Pygments                      2.16.1
pymongo                       4.2.0
pyOpenSSL                     21.0.0
pyparsing                     3.1.1
pyperclip                     1.8.2
pyrsistent                    0.20.0
pytest                        7.4.2
pytest-console-scripts        1.4.1
pytest-cov                    4.1.0
pytest-jupyter                0.7.0
pytest-timeout                2.2.0
pytest-tornasync              0.6.0.post2
python-dateutil               2.8.2
python-json-logger            2.0.7
pytz                          2021.1
pyxlsb                        1.0.10
PyYAML                        5.4.1
pyzmq                         25.1.1
referencing                   0.30.2
regex                         2024.4.16
requests                      2.31.0
requests-mock                 1.11.0
requests-oauthlib             1.3.1
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rich                          13.6.0
rpds-py                       0.10.6
rsa                           4.9
ruamel.yaml                   0.17.36
ruamel.yaml.clib              0.2.8
s3contents                    0.11.2
s3fs                          2021.11.0
s3transfer                    0.4.2
safetensors                   0.4.3
scikit-learn                  1.3.0
scipy                         1.12.0
SecretStorage                 3.3.3
seldon-core                   1.14.1
Send2Trash                    1.8.2
sentence-transformers         2.2.2
sentencepiece                 0.1.99
setuptools                    68.2.2
shellingham                   1.5.3
six                           1.16.0
sniffio                       1.3.0
snowballstemmer               2.2.0
soupsieve                     2.5
Sphinx                        7.1.2
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.1
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
sphinxcontrib-spelling        8.0.0
SQLAlchemy                    1.4.49
stack-data                    0.6.3
strict-rfc3339                0.7
sympy                         1.12
tenacity                      8.2.3
terminado                     0.17.1
threadloop                    1.0.2
threadpoolctl                 3.4.0
thrift                        0.20.0
tinycss2                      1.2.1
tokenizers                    0.19.1
tomli                         2.0.1
tomli_w                       1.0.0
tomlkit                       0.12.1
torch                         2.2.2
torchvision                   0.17.2
tornado                       6.3.3
tqdm                          4.66.1
traitlets                     5.11.2
transformers                  4.40.1
triton                        2.2.0
trove-classifiers             2023.10.18
typeguard                     2.12.1
types-python-dateutil         2.8.19.14
typing_extensions             4.8.0
typing-inspect                0.9.0
uri-template                  1.3.0
urllib3                       1.26.16
userpath                      1.9.1
uvicorn                       0.17.5
virtualenv                    20.24.5
wcwidth                       0.2.5
webcolors                     1.13
webencodings                  0.5.1
websocket-client              1.6.4
Werkzeug                      2.0.3
wrapt                         1.15.0
xlrd                          2.0.1
yarl                          1.9.2
zipp                          3.17.0
```

**附加信息 / Additional Information**
添加与问题相关的任何其他信息 / Add any other information related to the issue.

**Comment by tlevyaleia** [comment]
Please provide comment/answer in English!

**Comment by tlevyaleia** [comment]
I realize that I made a confusion between HuggingFaceHubEmbeddings and HuggingFaceHubEmbeddings classes.
In my case, I need to use HuggingFaceHubEmbeddings class that has a different signature.
I changed my code to using model_kwargs param instead :
```
from langchain_community.embeddings import HuggingFaceHubEmbeddings

model_kwargs = {'batch_size': 16}
embedding = HuggingFaceHubEmbeddings(model="my_url", huggingfacehub_api_token="xxx", model_kwargs=model_kwargs)

```
And I don't have an init error anymore.
But, I still have an issue with batch_size not taken into account when doing the embedding.
So, I will close this issue and raise an anoter one.
Thanks
